modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/leases/LeaseTracker.java

```diff
@@ -292,6 +321,29 @@ private CompletableFuture<Void> loadLeasesBusyAsync(long recoveryRevision) {
         return loadLeasesFuture;
     }
 
+    /**
+     * Fires the primary replica expire event if it needs.
+     *
+     * @param causalityToken Causality token.
+     * @param lease Lease to check on expiration.
+     */
+    private void firePrimaryReplicaExpiredEvenIfNeed(long causalityToken, Lease lease) {
+        if (isLeaseExpired(lease)) {
+            ReplicationGroupId grpId = lease.replicationGroupId();
+
+            Lease epiredLease = leases.leaseByGroupId().get(grpId);
+
+            CompletableFuture<Void> prev = expirationFutureByGroup.put(grpId, fireEvent(
+                    PRIMARY_REPLICA_EXPIRED,
+                    new PrimaryReplicaEventParameters(causalityToken, grpId, epiredLease.getLeaseholder())
+            ));
+
+            if (prev != null && !prev.isDone()) {
+                LOG.warn("Previous lease expiration process has not completed yet [grpId={}]", grpId);
```

Comment:
It could happens in the case when a expired, but the does not accepeted. I think again about it and desided to no trigger the expire event in the case.
Currently, we can change this warning message to just checkoing assertion.